### PR TITLE
Made reactive

### PR DESCRIPTION
**Pull request by gznznzjsn**

### CODE REVIEW

File: src/main/java/com/gznznzjsn/employeeservice/service/impl/EmployeeServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@Service
+@RequiredArgsConstructor
+public class EmployeeServiceImpl implements EmployeeService {
+
+    private final EmployeeRepository employeeRepository;
+
+    //    public Mono<ServerResponse> hello(ServerRequest request) {
```
Comment (h1alexbel) [inline src/main/java/com/gznznzjsn/employeeservice/service/impl/EmployeeServiceImpl.java:19]: remove comments

File: src/main/java/com/gznznzjsn/employeeservice/web/controller/EmployeeController.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/employee-api/v1/employees")
+public class EmployeeController {
+
+    private final EmployeeMapper employeeMapper;
+    private final EmployeeService employeeService;
+
+    //    @PreAuthorize("hasAuthority('EMPLOYEE_MANAGER')")
```
Comment (h1alexbel) [inline src/main/java/com/gznznzjsn/employeeservice/web/controller/EmployeeController.java:22]: comments

File: src/main/java/com/gznznzjsn/employeeservice/web/controller/EmployeeController.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    @GetMapping
+    public Flux<EmployeeDto> getAll() {
+        return employeeService.getAll().map(employeeMapper::toDto);
+    }
+
+    //    @PreAuthorize("hasAuthority('EMPLOYEE_MANAGER')")
```
Comment (h1alexbel) [inline src/main/java/com/gznznzjsn/employeeservice/web/controller/EmployeeController.java:28]: comments

File: src/main/java/com/gznznzjsn/employeeservice/web/controller/EmployeeController.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    @PostMapping
+    public Mono<EmployeeDto> create(@Validated(OnCreateEmployee.class) @RequestBody EmployeeDto employeeDto) {
+       return Mono.just(employeeDto)
+               .map(employeeMapper::toEntity)
+               .flatMap(employeeService::create)
+               .map(employeeMapper::toDto);
+    }
+
+    //    @PreAuthorize("hasAuthority('EMPLOYEE_MANAGER')")
```
Comment (h1alexbel) [inline src/main/java/com/gznznzjsn/employeeservice/web/controller/EmployeeController.java:37]: again comments

File: src/main/java/com/gznznzjsn/employeeservice/web/controller/EmployeeController.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    @GetMapping("/{employeeId}")
+    public Mono<EmployeeDto> get(@PathVariable Long employeeId) {
+        return employeeService
+                .get(employeeId)
+                .map(employeeMapper::toDto);
+    }
+
+    //    @PreAuthorize("hasAuthority('EMPLOYEE_MANAGER')")
```
Comment (h1alexbel) [inline src/main/java/com/gznznzjsn/employeeservice/web/controller/EmployeeController.java:45]: comments

**Review by h1alexbel — CHANGES_REQUESTED** [review]
@gznznzjsn remove comments, please

File: src/main/java/com/gznznzjsn/employeeservice/service/impl/EmployeeServiceImpl.java (hunk 1 above)
Comment (gznznzjsn) [reply src/main/java/com/gznznzjsn/employeeservice/service/impl/EmployeeServiceImpl.java:19]: done

**Review by h1alexbel — APPROVED** [review]

**Comment by h1alexbel** [comment]
@gznznzjsn, please name your pull request properly!

**Pull request merged**